src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -942,97 +901,44 @@ public List<Pool> getBySubscriptionIds(String ownerId, Collection<String> subIds
-        Integer result = (Integer) currentSession()
-            .createCriteria(Entitlement.class)
-            .setProjection(Projections.rowCount())
-            .createCriteria("pool")
-            .createCriteria("sourceEntitlement")
-            .add(Restrictions.idEq(entitlementId))
-            .list()
-            .get(0);
-
-        enableIfPrevEnabled(consumerFilter);
-        return result;
-    }
+        TypedQuery<Pool> query = getEntityManager().createQuery(jpql, Pool.class)
+            .setParameter("ownerId", ownerId);
 
-    // TODO: watch out for performance. Should we limit the certificates
-    // retrieved?
-    @SuppressWarnings("unchecked")
-    public List<EntitlementCertificate> retrieveEntCertsOfPoolsWithSourceEntitlement(String entId) {
-        return currentSession().createCriteria(EntitlementCertificate.class)
-            .createCriteria("entitlement")
-            .createCriteria("pool")
-            .createCriteria("sourceEntitlement")
-            .add(Restrictions.idEq(entId))
-            .list();
-    }
+        for (Map<String, Entitlement> block: this.partitionMap(subIdMap)) {
+            query.setParameter("subscriptionIds", new ArrayList<>(block.keySet()));
+            query.setParameter("entitlements", new ArrayList<>(block.values()));
 
-    /**
-     * @param consumerFilter
-     */
-    private void enableIfPrevEnabled(Filter consumerFilter) {
-        // if filter was previously enabled, restore it.
-        if (consumerFilter != null) {
-            FilterImpl filterImpl = (FilterImpl) consumerFilter;
-            Filter filter = currentSession().enableFilter(CONSUMER_FILTER);
-            filter.setParameter("consumer_id", filterImpl.getParameter("consumer_id"));
+            result.addAll(query.getResultList());
         }
-    }
 
-    public Filter disableConsumerFilter() {
-        Filter consumerFilter = currentSession().getEnabledFilter(CONSUMER_FILTER);
-        currentSession().disableFilter(CONSUMER_FILTER);
-        return consumerFilter;
+        return result;
     }
 
-    public List<ActivationKey> getActivationKeysForPool(Pool p) {
-        List<ActivationKey> activationKeys = new ArrayList<>();
-        List<ActivationKeyPool> activationKeyPools = currentSession().createCriteria(
-            ActivationKeyPool.class).add(Restrictions.eq("pool", p)).list();
+    public List<ActivationKey> getActivationKeysForPool(Pool pool) {
+        String jpql = """
+            SELECT akp FROM ActivationKeyPool akp
+            WHERE akp.pool = :pool""";
 
-        for (ActivationKeyPool akp : activationKeyPools) {
-            activationKeys.add(akp.getKey());
-        }
+        List<ActivationKeyPool> activationKeyPools = getEntityManager()
+            .createQuery(jpql, ActivationKeyPool.class)
+            .setParameter("pool", pool)
+            .getResultList();
 
-        return activationKeys;
+        return activationKeyPools.stream()
```

Comment:
For this, we can go right into the stream and combine it with the previous line:

```
this.getEntityManager()
  .createQuery(jpql, ActivationKeyPool.class)
  .setParameter("pool", pool)
  .getResultStream() // return a stream instead of list
  .map(ActivationKeyPool::getKey)
  .toList();
```

Reply:
You can not use getResultStream() on TypedQuery/Query class, but i just picked the ActivationKey in hql

Reply:
Oh is that method not available in the version of the JPA interface we have available? Unfortunate. I guess when we move to the versions where it's got the Jakarta namespace we can start using it.

But good call on this change. It'll pull less data from the DB to throw out, so it'll likely be more performant anyway.